packages/net-stubbing/lib/server/middleware/request.ts

```diff
@@ -32,7 +32,7 @@ export const SetMatchingRoutes: RequestMiddleware = async function () {
 
   // if this is a request to the dev server, do not match any routes as
   // we do not want to allow the user to intercept requests to the dev server
-  if (url.pathname.startsWith(this.config.devServerPublicPathRoute)) {
+  if (url.pathname.startsWith('/__cypress/src')) {
```

Comment:
I am not sure if we want to hard code this here since a user could also have the default just configured, which makes the default different. What is the use case for intercepting requests to the dev server?

Reply:
I agree - `cy.intercept` is more for requests to your **app server** - I don't think this fix makes sense. I guess you could intercept a ES module request, or something, and return a mock module, or a CSS asset, but again this seems strange - you'd be testing a development only behaviour, and those requests might change or be different in the production bundle.

I too am curious on the OP's use case, there might be a better way to accomplish some of these things.